### ISSUE TITLE
semgrep returns 0 exit code when configuration path doesn't exist

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [semgrep/semgrep#7891]
**Describe the bug**

Semgrep return exits with 0 when the config path is not found

**To Reproduce**

```bash
$ docker run -it returntocorp/semgrep semgrep ci --config nonexisting.yaml
                  
                  
┌────────────────┐
│ Debugging Info │
└────────────────┘
                  
  SCAN ENVIRONMENT
  versions    - semgrep 1.22.0 on python 3.11.3                        
  environment - running in environment git, triggering event is unknown
[ERROR] WARNING: unable to find a config; path `nonexisting.yaml` does not exist (since you are running in docker, you cannot specify arbitrary paths on the host; they must be mounted into the container)
[ERROR] invalid configuration file found (1 configs were invalid)
$ echo $?
0
```

**Expected behavior**
 
It should return 1 so users can catch this error.

**Screenshots**

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [X] P2: regular bug that should get fixed

**Environment**

docker

**Use case**

Explicitly failing when the configuration file is not found to avoid things like https://github.com/inspektor-gadget/inspektor-gadget/pull/1671.

ref https://github.com/returntocorp/semgrep/issues/7630

**Comment by eiffel-fl** [comment]
Hi.

Searching a bit in the code base, I figured out errors are suppressed as default:
https://github.com/returntocorp/semgrep/blob/11f9013619c081f1fdf4f680fd4521eca826f161/cli/src/semgrep/commands/ci.py#L150-L162

So, calling the above with `--no-suppress-errors` results in a non 0 exit code:

```
$ docker run -it returntocorp/semgrep semgrep ci --config nonexisting.yaml --no-suppress-errors                      
                  
                  
┌────────────────┐
│ Debugging Info │
└────────────────┘
                  
  SCAN ENVIRONMENT
  versions    - semgrep 1.23.0 on python 3.11.3                        
  environment - running in environment git, triggering event is unknown
[ERROR] WARNING: unable to find a config; path `nonexisting.yaml` does not exist (since you are running in docker, you cannot specify arbitrary paths on the host; they must be mounted into the container)
[ERROR] invalid configuration file found (1 configs were invalid)
$ echo $?                                                                                                            
7
```

Now, I do not really understand the why of this?
Particularly because calling `semgrep` with `cli` instead of `ci` leads to having a non zero exit code:

```
$ docker run -it returntocorp/semgrep semgrep cli --config nonexisting.yaml 
                  
                  
┌────────────────┐
│ Debugging Info │
└────────────────┘
                  
  SCAN ENVIRONMENT
  versions    - semgrep 1.23.0 on python 3.11.3                        
  environment - running in environment git, triggering event is unknown
[ERROR] WARNING: unable to find a config; path `nonexisting.yaml` does not exist (since you are running in docker, you cannot specify arbitrary paths on the host; they must be mounted into the container)
[ERROR] invalid configuration file found (1 configs were invalid)
$ echo $?                                                                                                            
7
```

Do we need to change the way we call `semgrep` in our CI to add `--no-suppress-errors`?


Best regards and thank you in advance.

**Comment by mauriciovasquezbernal** [comment]
@eiffel-fl using `--no-suppress-errors` in our pipeline will fix the issue, however I found it weird that `--supress-errors` is `True` by default, btw it was changed in https://github.com/returntocorp/semgrep/pull/5886. I suppose that there are good reasons for this change and that this issue can be closed? I'll let it open but if any mantainer wants to close it that's fine for me.

**Comment by IagoAbal** [comment]
I didn't realize myself this was `semgrep ci` and yeah we have some "fail open" by default. It's documented here: https://semgrep.dev/docs/semgrep-ci/running-semgrep-ci-without-semgrep-cloud-platform/#passing-or-failing-the-ci-job

> Fail on blocking findings, but passes on internal errors. This is the default behavior.

This is so that, for example, bugs in Semgrep don't block PRs.

We should probably print some message in the output clarifying this behavior and pointing to `--no-suppress-errors`.

**Comment by eiffel-fl** [comment]
Hi.


> bugs in Semgrep don't block PRs.

This definitely makes sense! Thank you for the precision!

> We should probably print some message in the output clarifying this behavior and pointing to --no-suppress-errors.

What type of message would you like and would you welcome a PR adding this?


Best regards.

**Comment by IagoAbal** [comment]
> What type of message would you like and would you welcome a PR adding this?

Don't know, something like "There were errors during analysis but Semgrep will succeed because there were no blocking findings, use --no-suppress-errors if you want Semgrep to fail when there are errors."

**Comment by adamboros** [comment]
This seems to be fixed by #8069, closing. Thank you for the contribution!
